scripts/west_commands/footprint.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+import argparse
+import os
+import pprint
+import re
+from pathlib import Path
+import subprocess
+
+'''west "footprint" command'''
+
+from west.commands import WestCommand
+from runners.core import BuildConfiguration
+from build_helpers import find_build_dir, is_zephyr_build, \
+    FIND_BUILD_DIR_DESCRIPTION
+from run_common import desc_common, add_parser_common, do_run_common
+
+from zcmake import CMakeCache
+from zephyr_ext_common import Forceable, cached_runner_config, \
+    zephyr_scripts_path
+
+
+class Footprint(Forceable):
```

Comment:
Very interesting, thanks for sharing.

To leave room for other tools, would be nice to see this called 'Puncover' and the command 'puncover' instead.

Reply:
I would leave it as footprint and provide the option to support multiple tools, a command name "puncover" does not really mean anything to many people :-D
Anyways, we will see...

Reply:
Yes, I agree with @nashif. We can default to a particular tool, but the actual command should be called `footprint` IMHO.

Reply:
I have a lot of concerns about this approach. Extensions are cheap, why not keep the code separate?

Reply:
well, following this approach would mean that we should have 'west pycocd' or 'west jlink' instead of 'west flash', no?

Reply:
If you can make the footprint command output nice and generic depending on what the user has installed on their system then I take your point.